### ISSUE TITLE
Conflict between @freezed and @JsonSerializable() annotations

### ISSUE THREAD

**Issue by aliazimoshan** [rrousselGit/freezed#993]
**Describe the bug**
I'm facing an issue when combining @freezed and @JsonSerializable() annotations on the same class. The generated files include fromJson/toJson methods that may conflict, resulting in an error.

**To Reproduce**

1.  Create a class using both @freezed and @JsonSerializable() annotations.
2.  Notice that the generated files include conflicting fromJson/toJson methods.

```
@JsonSerializable()
@freezed
abstract class MyModel with _$MyModel {
  const factory MyModel({
    // ... fields ...
  }) = _MyModel;

  factory MyModel.fromJson(Map<String, dynamic> json) => _$MyModelFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$MyModelToJson(this);
}
```
### Error Message

> The name '$CallHistoryModelFromJson' is already defined.
Try renaming one of the declarations.dartduplicate_definition
call_history_model.freezed.dart(17, 18): The first definition of this name.
The declaration '$CallHistoryModelFromJson' isn't referenced.
Try removing the declaration of '_$CallHistoryModelFromJson'.

**Expected behavior**
I expect to be able to use @freezed and @JsonSerializable() together without conflicts in the generated code.

**Comment by rrousselGit** [comment]
You should not specify `@JsonSerializable` on the freezed class.

Freezed generates fromJson/toJson on its own.
If you want to customise generation, the syntax is:


```
@freezed
abstract class MyModel with _$MyModel {
  @JsonSerializable(...)
  const factory MyModel({
    // ... fields ...
  }) = _MyModel;

  factory MyModel.fromJson(Map<String, dynamic> json) => _$MyModelFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$MyModelToJson(this);
}
```

**Comment by aliazimoshan** [comment]
Thank you for your prompt response! I'm happy to report that the issue has been resolved.

The root cause was related to storing data in a sembast database using a freezed-generated model. The crucial realization that led to the solution was pinpointing an incorrect placement of the @JsonSerializable() annotation within the freezed model.

```
@freezed
abstract class MyModel with _$MyModel {
  @JsonSerializable(explicitToJson: true)
  const factory MyModel({
    // ... fields ...
  }) = _MyModel;

  factory MyModel.fromJson(Map<String, dynamic> json) => _$MyModelFromJson(json);

}
```

I also found valuable insights in the solution shared in this [GitHub comment](https://github.com/tekartik/sembast.dart/issues/254#issuecomment-855264076), which reinforced my understanding of the issue.

Thank you once again for your assistance and the provided solution link!
